File: HackerRank/Find Digits/Find Digits.py
t=int(input())
for tc in range(t):
    n=input()
    z=list(map(int,list(n)))    # Making a list where it's elements are the digits of that number
    count=0
    for i in z:
        if i>0 and int(n)%i==0:     # We will exclude the digit '0' as it can't be a divisor
            count+=1    # add 1 to our count if a digit is a divisor
    print(count)

File: HackerRank/Find Digits/Find Digits.java
import java.util.*;
public class findDigits {


	public static int Digits(int n) {

		int m=n, count=0;
		while(n>0){
			int p=n%10;                                   //Takes the last digit out of the number.
			if(p!=0 && m%p==0)                            //Checking if the digit completely divides the number. Division by 0 is not allowed.
				count++;  
			n/=10;          
		}
		return count;                                     //Returns the number of digits that divide the number.
	}

	public static void main(String[] args) {
		Scanner sc= new Scanner(System.in);
		int T= sc.nextInt(); 
		for(int t=1; t<=T; t++) {
			int n= sc.nextInt();
			System.out.println(Digits(n));
		}
		sc.close();
	}
}
